Add oAuth2 functionality

davidkokkelink opened the pull request:
This PR adds the third-party login functionality to Datawrapper's frontend service. Some architectural decisions:

- the code uses the [`@hapi/bell` plugin](https://github.com/hapijs/bell) to handle the oAuth2 process.
- it's entirely **provider-agnostic**, and fetches its available providers from the `config`:
```js
oauth: {
    twitter: {
        id: "abc",
        secret: "xyz"
    },
    google: {
        id: "def",
        secret: "uvw"
    },
    // ...
}
```
- the `frontend` registers the auth strategy and the `/signin/:provider` endpoint. once that endpoint was called, a regular user session is used so no changes are needed on other services.

---

I briefly tested the oAuth2-flow on production over the weekend (with Github, Google and twitter) and things worked as expected (difficult to impossible to do locally due to the `redirect_uri`s, etc.)

### Launch checklist
- [x] Change the `redirect_uri` in the twitter app to use `https`

gka commented:
Really cool that this is provider agnostic now. Two thoughts:

I would move the config into `config.general.oauth`, or maybe just `config.frontend.oauth` in case users can only login to the frontend service using OAuth. 

Let's reserve first level config options for services.

Second: in case someone adds garbage keys to the `oauth` config section I think we should check the provider id against some form of whitelist (which I'm sure the `bell` plugin must have somewhere)

davidkokkelink commented:
Thanks for the suggestions. I moved the config into `general.oauth` (it's easy to foresee scenarios where oAuth in some way is relevant for the api as well, so `general` seemed more sensible than `frontend`).

Also, I added a check based on `bell`'s provider list and log a warning if it's not supported.
